System/Random/Internal.hs

```diff
@@ -440,31 +448,93 @@ class Uniform a where
 -- > uniformRM (a,b) = uniformM (b,a)
 --
 -- @since 1.2
-class UniformRange a where
-  uniformRM :: MonadRandom g s m => (a, a) -> g s -> m a
+class UniformRange (l :: Clusivity) (u :: Clusivity) a where
+  uniformRM :: MonadRandom g s m => (Bound l a, Bound u a) -> g s -> m a
 
-instance UniformRange Integer where
+data Clusivity = Inclusive | Exclusive
+
+type family Bound (c :: Clusivity) = r | r -> c where
+  Bound 'Inclusive = Inc
+  Bound 'Exclusive = Exc
+
+newtype Inc a =
+  Inc { unInc :: a }
+  deriving (Num, Show, Eq, Ord, Integral, Floating, Real, Fractional, Bounded, Enum)
+
+newtype Exc a =
+  Exc { unExc :: a }
+  deriving (Num, Show, Eq, Ord, Integral, Floating, Real, Fractional, Bounded, Enum)
+
+
+
+
+uniformIntegralExcExcRM ::
+     (MonadRandom g s m, UniformRange 'Inclusive 'Inclusive a, Integral a)
+  => (Bound 'Exclusive a, Bound 'Exclusive a)
+  -> g s
+  -> m a
+uniformIntegralExcExcRM (Exc l, Exc u) = uniformRM (Inc (l + 1), Inc (u - 1))
```

Comment:
This function assumes that l < u while old uniformRM implementation ensured that `uniformRM (a,b) = uniformRM (b,a)`. Not to mention possible over/underflow problems

Reply:
I am glad you caught it, but I left it out on purpose, because it is a separate issue, see my comment: https://github.com/idontgetoutmuch/random/issues/113#issuecomment-620147980